packages/dm-core/src/components/UiRecipesSideBarSelector.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useContext, useEffect, useState } from 'react'
+
+import styled from 'styled-components'
+import { CircularProgress } from '@equinor/eds-core-react'
+import { UiPluginContext } from '../context/UiPluginContext'
+import { ErrorBoundary } from '../utils/ErrorBoundary'
+import { useBlueprint } from '../hooks'
+import { IUIPlugin } from '../types'
+
+const lightGray = '#d3d3d3'
+
+const RecipeSidebarWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  padding-right: 5px;
+  border-right: 2px black solid;
+`
+
+const Wrapper = styled.div`
+  align-self: start;
+  width: 100%;
+  display: flex;
+`
+
+interface ISPButton {
+  active: boolean
+}
+
+const SelectPluginButton = styled.div<ISPButton>`
+  padding: 5px 10px;
+  margin: 10px;
+  text-align: center;
+  border-bottom: ${(props: any) =>
+    (props.active == true && '2px #017078FF solid') ||
+    `2px ${lightGray} solid`};
+  &:hover {
+    background: ${lightGray};
+    cursor: pointer;
+  }
+`
+
+type TSelectablePlugins = {
+  name: string
+  component: (props: IUIPlugin) => JSX.Element
+  config: any
+}
+
+export function UiRecipesSideBarSelector(props: IUIPlugin): JSX.Element {
+  const { idReference, categories, onSubmit, onOpen, type } = props
+  const { loading, getUiPlugin } = useContext(UiPluginContext)
+  const { uiRecipes, isLoading } = useBlueprint(type)
+  const [selectedRecipe, setSelectedRecipe] = useState<number>(0)
+  const [selectableRecipes, setSelectableRecipes] = useState<
+    TSelectablePlugins[]
+  >([])
+
+  useEffect(() => {
+    // Make sure ui plugins have been loaded
+    if (loading || isLoading) return
+    const recipes = uiRecipes.map((uiRecipe: any) => ({
+      name: uiRecipe?.label || uiRecipe?.name || uiRecipe?.plugin || 'No name',
+      component: getUiPlugin(uiRecipe?.plugin).component,
+      config: uiRecipe?.config,
+    }))
+    // @ts-ignore
```

Comment:
can this ts-ignore be deleted if we add a return type to the function inside the above map function?


```JavaScript
    const recipes = uiRecipes.map((uiRecipe: any): TSelectablePlugins => ({
      name: uiRecipe?.label || uiRecipe?.name || uiRecipe?.plugin || 'No name',
      component: getUiPlugin(uiRecipe?.plugin).component,
      config: uiRecipe?.config,
    }))
```